src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.Values.StringSegment.cs

```diff
@@ -941,146 +1245,191 @@ public static void WriteStringValueSegment_Empty()
-        private static void WriteStringValueHelper<T>(Utf8JsonWriter writer, ReadOnlySpan<T> value)
-            where T : struct
-        {
-            if (typeof(T) == typeof(char))
             {
-                writer.WriteStringValue(MemoryMarshal.Cast<T, char>(value));
-            }
-            else if (typeof(T) == typeof(byte))
-            {
-                writer.WriteStringValue(MemoryMarshal.Cast<T, byte>(value));
+                var output = new ArrayBufferWriter<byte>();
+                using var jsonUtf8 = new Utf8JsonWriter(output);
+                jsonUtf8.WriteBase64StringSegment([], true);
+                jsonUtf8.Flush();
+                JsonTestHelper.AssertContents("\"\"", output);
             }
-            else
+
             {
-                if (typeof(T) == typeof(int))
-                {
-                    Assert.Fail($"Did you pass in int or int[] instead of byte or byte[]? Type {typeof(T)} is not supported by {nameof(WriteStringValueHelper)}.");
-                }
-                else
-                {
-                    Assert.Fail($"Type {typeof(T)} is not supported by {nameof(WriteStringValueHelper)}.");
-                }
+                var output = new ArrayBufferWriter<byte>();
+                using var jsonUtf8 = new Utf8JsonWriter(output);
+                jsonUtf8.WriteBase64StringSegment([], false);
+                jsonUtf8.Flush();
+                JsonTestHelper.AssertContents("\"", output);
             }
-        }
 
-        // Switch this to use an enum discriminator input when base64 is supported
-        private static void WriteStringValueSegmentHelper<T>(Utf8JsonWriter writer, ReadOnlySpan<T> value, bool isFinal)
-            where T : struct
-        {
-            if (typeof(T) == typeof(char))
             {
-                writer.WriteStringValueSegment(MemoryMarshal.Cast<T, char>(value), isFinal);
+                var output = new ArrayBufferWriter<byte>();
+                using var jsonUtf8 = new Utf8JsonWriter(output);
+                jsonUtf8.WriteBase64StringSegment([], false);
+                jsonUtf8.WriteBase64StringSegment([], true);
+                jsonUtf8.Flush();
+                JsonTestHelper.AssertContents("\"\"", output);
             }
-            else if (typeof(T) == typeof(byte))
+
             {
-                writer.WriteStringValueSegment(MemoryMarshal.Cast<T, byte>(value), isFinal);
+                var output = new ArrayBufferWriter<byte>();
+                using var jsonUtf8 = new Utf8JsonWriter(output);
+                jsonUtf8.WriteBase64StringSegment([], false);
+                jsonUtf8.WriteBase64StringSegment([], false);
+                jsonUtf8.WriteBase64StringSegment([], true);
+                jsonUtf8.Flush();
+                JsonTestHelper.AssertContents("\"\"", output);
             }
-            else
+        }
+
+        enum EncodingType
```

Comment:
```suggestion
        enum PartialStringEncodingType
```

Reply:
Changed to `StringValueEncodingType` since this also is used as a helper for the non-segmented API.